README.md

```diff
@@ -148,3 +148,67 @@ set the logger level to INFO:
 
     logging.getLogger('backoff').setLevel(logging.INFO)
 
+## Examples for Full Jitter and Equal Jitter
+
+*Full Jitter and Equal Jitter algorithm comes from [AWS Blog](http://www.awsarchitectureblog.com/2015/03/backoff.html), basically usage is almost identical to aforementioned examples, hence the unique difference would be described in this session.*
+
+### @backoff.on_exception
+
+The ``on_exception`` decorator is used to retry when a specified exception is raised. Here's an example using AWS exponential backoff when any requests exception is raised:
+
+    @backoff.on_exception(backoff.expo,
+                          jitter=backoff.full_jitter,
+                          requests.exceptions.RequestException,
+                          max_tries=8)
+    def get_url(url):
+        return requests.get(url)
+
+To take advantage of Full Jitter, you may just specify ``backoff.expo``, ``jitter=backoff.full_jitter`` and it should work as expected.
+
+## Make sure you are ready to 'commit'
+
+### Virtual Python Environment builder
+
+``virtualenv`` is a tool to create isolated testing environments which could prevent from pollution.
+
+	# Install virtualenv
+	$ sudo pip install virtualenv
+	
+	# Create virtual environment
+	$ virtualenv mytest
+	
+	# Activate virtual environment
+	$ cd mytest
+	$ source bin/activate
+
+### Python style guide checker
+
+    (mytest)$ sudo pip install pep8
```

Comment:
Two things here:
1. You shouldn't need to and it's preferable not to `sudo` when installing packages into a virtualenv environment.
2. There is a requirements.txt file in the repo which defines these deps. You can simplify these instructions by recommending:

    pip install -r requirements.txt